src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/X509AuthorityKeyIdentifierExtension.cs

```diff
@@ -0,0 +1,575 @@
+            }
+
+            return new X509AuthorityKeyIdentifierExtension(encoded);
+        }
+
+        /// <summary>
+        ///   Creates an <see cref="X509AuthorityKeyIdentifierExtension"/> that specifies
+        ///   the provided issuer name and serial number.
+        /// </summary>
+        /// <param name="issuerName">
+        ///   The issuer name value from the Certificate Authority (CA) certificate that will
+        ///   sign this extension.
+        /// </param>
+        /// <param name="serialNumber">
+        ///   The serial number value from the Certificate Authority (CA) certificate that will
+        ///   sign this extension.
+        /// </param>
+        /// <returns>
+        ///   The configured extension.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        ///   <paramref name="issuerName"/> or <paramref name="serialNumber"/> is <see langword="null" />.
+        /// </exception>
+        public static X509AuthorityKeyIdentifierExtension CreateFromIssuerNameAndSerialNumber(
+            X500DistinguishedName issuerName,
+            byte[] serialNumber)
+        {
+            ArgumentNullException.ThrowIfNull(issuerName);
+            ArgumentNullException.ThrowIfNull(serialNumber);
+
+            return CreateFromIssuerNameAndSerialNumber(issuerName, new ReadOnlySpan<byte>(serialNumber));
+        }
+
+        /// <summary>
+        ///   Creates an <see cref="X509AuthorityKeyIdentifierExtension"/> that specifies
+        ///   the provided issuer name and serial number.
+        /// </summary>
+        /// <param name="issuerName">
+        ///   The issuer name value from the Certificate Authority (CA) certificate that will
+        ///   sign this extension.
+        /// </param>
+        /// <param name="serialNumber">
+        ///   The serial number value from the Certificate Authority (CA) certificate that will
+        ///   sign this extension.
+        /// </param>
+        /// <returns>
+        ///   The configured extension.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        ///   <paramref name="issuerName"/> is <see langword="null" />.
+        /// </exception>
+        public static X509AuthorityKeyIdentifierExtension CreateFromIssuerNameAndSerialNumber(
+            X500DistinguishedName issuerName,
+            ReadOnlySpan<byte> serialNumber)
+        {
+            ArgumentNullException.ThrowIfNull(issuerName);
+
+            AsnWriter writer = new AsnWriter(AsnEncodingRules.DER);
+
+            using (writer.PushSequence())
+            {
+                using (writer.PushSequence(new Asn1Tag(TagClass.ContextSpecific, 1)))
+                using (writer.PushSequence(new Asn1Tag(TagClass.ContextSpecific, 4)))
+                {
+                    writer.WriteEncodedValue(issuerName.RawData);
+                }
+
+                writer.WriteIntegerUnsigned(serialNumber, new Asn1Tag(TagClass.ContextSpecific, 2));
```

Comment:
I'm not sure how much we care, but this relies on the encoder to validate the integer. For example, this snippet:

```C#
X509AuthorityKeyIdentifierExtension.CreateFromIssuerNameAndSerialNumber(
    new X500DistinguishedName("CN=bad serial"),
    new byte[8]);
```

will throw a `ArgumentException` from `WriteIntegerUnsigned`. Do we want that exception to escape, or handle it some other way?

Reply:
Alternatively, could consider garbage-in-garbage-out for the serial number. If someone has a malformed serial number... chances are they are going to want to use it anyway in the extension. "Yes my CA has a garbage subject key and I want my AKI to identify my garbage".

Reply:
Garbage-In/Garbage-Out is hard, since there's not a good bypass in the AsnWriter, and that CA is already on shaky grounds for interop.  Rather than do something like CertificateRequest's NormalizeSerialNumber (which helps let someone do RNG.Fill and not have to worry about it) I think rethrowing it with a corrected parameter name is probably the best bet.

Hm.

Reply:
It occurs to me that WriteIntegerUnsigned isn't correct, it should just be WriteInteger.

Some tool (Windows certreq, maybe?) historically forgot to add padding bytes (or clear the sign bit) and emitted negative serial numbers; and this shouldn't rewrite them to be positive.

Reply:
Hmmm.... Actually.  It's a `[2] IMPLICIT INTEGER`, not an integer directly.  I could cheat in the writer and call it `[4] IMPLICIT OCTET STRING`.  But I think I'll go with the ArgumentException.